packages/angular_devkit/build_webpack/src/angular-cli-files/models/webpack-configs/browser.ts

```diff
@@ -21,34 +20,29 @@ import { WebpackConfigOptions } from '../build-options';
 export function getBrowserConfig(wco: WebpackConfigOptions) {
   const { root, projectRoot, buildOptions, appConfig } = wco;
 
-
-  let extraPlugins: any[] = [];
+  const extraPlugins: any[] = [];
 
   // figure out which are the lazy loaded entry points
   const lazyChunks = lazyChunksFilter([
     ...extraEntryParser(appConfig.scripts, root, 'scripts'),
     ...extraEntryParser(appConfig.styles, root, 'styles')
   ]);
 
-  // TODO: Enable this once HtmlWebpackPlugin supports Webpack 4
-  const generateIndexHtml = false;
-  if (generateIndexHtml) {
-    extraPlugins.push(new HtmlWebpackPlugin({
-      template: path.resolve(root, appConfig.index),
-      filename: path.resolve(buildOptions.outputPath, appConfig.index),
-      chunksSortMode: packageChunkSort(appConfig),
-      excludeChunks: lazyChunks,
-      xhtml: true,
-      minify: buildOptions.optimization ? {
-        caseSensitive: true,
-        collapseWhitespace: true,
-        keepClosingSlash: true
-      } : false
-    }));
-    extraPlugins.push(new BaseHrefWebpackPlugin({
-      baseHref: buildOptions.baseHref as string
-    }));
-  }
+  extraPlugins.push(new HtmlWebpackPlugin({
+    template: path.resolve(root, appConfig.index),
```

Comment:
was just looking to submit my own similar patch. Would it be possible to update this to allow for a given `appConfig.template` entry? 
```
template: path.resolve(root, appConfig.template || appConfig.index), 
```
Allows for an index template override, with a fallback to the current index entry...

Reply:
In all fairness I am still looking into making the CI green as I still have some tests failing.

Not quite understanding why you need two templates. If you want to override the index file. You can specify this in the cli config. by changing the `index.html` to something else.